libs/langchain/langchain/llms/bedrock.py

```diff
@@ -34,17 +34,17 @@ def prepare_input(
         return input_body
 
     @classmethod
-    def prepare_output(cls, provider: str, response: Any) -> str:
+    def prepare_output(cls, provider: str, response: Any) -> Tuple(str, bool):
```

Comment:
Or maybe this can be decomposed into 3 separate functions: `parse_request_body` `prepare_output` `detect_completion`. Might be cleaner.

But the core functionality of prepare_output needs to be broken I think. Otherwise, we loose out on ways to detect completion.

Reply:
Also fyi this and streaming response have different response body structures.